presentation/src/main/java/me/androidbox/presentation/agenda/screen/AgendaScreen.kt

```diff
@@ -160,44 +160,66 @@ fun AgendaScreen(
+                                AgendaType.EVENT -> AgendaCardType.EVENT
+                                AgendaType.TASK -> AgendaCardType.TASK
+                                AgendaType.REMINDER -> AgendaCardType.REMINDER
+                            },
+                            isAgendaCompleted = false,
+                            onMenuOptionClicked = {
+                                println("Event ${agendaItem.id} has been clicked")
+                                agendaScreenEvent(AgendaScreenEvent.OnAgendaItemClicked(agendaItem))
+                                agendaScreenEvent(
+                                    AgendaScreenEvent.OnChangeShowEditAgendaItemDropdownStatus(
+                                        shouldOpen = true,
+                                        agendaItem = agendaItem
+                                    )
+                                )
+                            },
+                            dropDownMenu = {
+                                /** Open, Edit, Delete Agenda Items */
+                                AgendaDropDownMenu(
+                                    modifier = Modifier
+                                        .background(color = MaterialTheme.colorScheme.dropDownMenuBackgroundColor)
+                                        .align(Alignment.BottomEnd),
+                                    shouldOpenDropdown = agendaItem == agendaScreenState.agendaItemWithOperationBeingSelected && agendaScreenState.shouldOpenEditAgendaDropdown,
+                                    onCloseDropdown = {
+                                        agendaScreenEvent(
+                                            AgendaScreenEvent.OnChangeShowEditAgendaItemDropdownStatus(
+                                                shouldOpen = false,
+                                                agendaItem = agendaItem
+                                            )
+                                        )
+                                    },
+                                    listOfMenuItemId = AgendaMenuActionType.values()
+                                        .map { it.titleId },
+                                    onSelectedOption = { item ->
+                                        agendaScreenState.agendaItemClicked?.let { agendaItem ->
+                                            onSelectedEditAgendaItemClicked(
+                                                agendaItem,
+                                                AgendaMenuActionType.values()[item]
+                                            )
+                                            agendaScreenEvent(
+                                                AgendaScreenEvent.OnChangeShowEditAgendaItemDropdownStatus(
+                                                    shouldOpen = false,
+                                                    agendaItem = agendaItem
```

Comment:
If you are closing the dropdown, why do you need to pass an agendaitem?

Reply:
@MKiperszmid I need to pass the agenda item as that is the agenda item that is associated with that dropdownmenu. As each one has its own dropdownmenu, it needs to know which one it will show/hide.

Reply:
No, this shouldn't be necessary as you only need to set the agenda item the drop down is associated with to null in order to hide the menu